p2p/src/peer_manager/tests/connections.rs

```diff
@@ -875,15 +873,19 @@ where
 
     // All peers should discover each other
     loop {
-        let connected_peers_1 = get_connected_peers(&tx1).await.len();
-        let connected_peers_2 = get_connected_peers(&tx2).await.len();
-        let connected_peers_3 = get_connected_peers(&tx3).await.len();
-
-        const EXPECTED_COUNT: usize = 4;
-        if connected_peers_1 == EXPECTED_COUNT
-            && connected_peers_2 == EXPECTED_COUNT
-            && connected_peers_3 == EXPECTED_COUNT
-        {
+        let counts = [
+            get_connected_peers(&tx1).await.len(),
+            get_connected_peers(&tx2).await.len(),
+            get_connected_peers(&tx3).await.len(),
```

Comment:
Not sure if this is intentional or not, but you can also use join! to await all of them at once instead of one by one.